example/lib/screens/preference_center.dart

```diff
@@ -0,0 +1,250 @@
+
+  PreferenceCenterConfig preferenceCenterConfig;
+  List<String> activeChannelSubscriptions;
+  Map<String, List<String>> activeContactSubscriptions;
+
+  @override
+  void initState() {
+    initAirshipListeners();
+    updatePreferenceCenterConfig();
+    fillInSubscriptionList();
+    Airship.trackScreen('Prefrence Center');
+    super.initState();
+  }
+
+  // Platform messages are asynchronous, so we initialize in an async method.
+  Future<void> initAirshipListeners() async {
+    Airship.onShowPreferenceCenter.listen((event) {
+    });
+  }
+
+  void fillInSubscriptionList() async {
+    SubscriptionList subscriptionList = await Airship.getSubscriptionLists(["channel", "contact"]);
+    activeChannelSubscriptions = subscriptionList.channelSubscriptionLists;
+    var contactSubscriptionLists = subscriptionList.contactSubscriptionLists;
+    for (ContactSubscriptionList contact in contactSubscriptionLists) {
+      activeContactSubscriptions[contact.identifier] = contact.scopes;
+    }
+  }
+  bool isSubscribedChannelSubscription (String subscriptionId) {
+    if (activeChannelSubscriptions != null) {
+      return activeChannelSubscriptions.contains(subscriptionId);
+    }
+    return false;
+  }
+
+  bool isSubscribedContactSubscription (String subscriptionId, List<String> scopes) {
+    print('MounaLog activeContactSubscriptions $activeContactSubscriptions');
+    if (activeContactSubscriptions != null) {
+      if (scopes.isEmpty) {
+        return activeContactSubscriptions.containsKey(subscriptionId);
+      }
+      List<String> activeContactSubscriptionsScopes = activeContactSubscriptions[subscriptionId];
+      return activeContactSubscriptionsScopes.contains(scopes);
+    }
+    return false;
+  }
+
+  Future updatePreferenceCenterConfig() async {
+    PreferenceCenterConfig config = await Airship.getPreferenceCenterConfig("neat");
+    setState(() {
+      preferenceCenterConfig = config;
+    });
+  }
+
+  void onPreferenceChannelItemToggled(String subscriptionId, bool isSubscribed) {
+    SubscriptionListEditor editor = Airship.editChannelSubscriptionLists();
+    if (isSubscribed) {
+      editor.subscribe(subscriptionId);
+      activeChannelSubscriptions.add(subscriptionId);
+    } else {
+      editor.unsubscribe(subscriptionId);
+      activeChannelSubscriptions.remove(subscriptionId);
+    }
+    editor.apply();
+    setState(() {});
+  }
+
+  void onPreferenceContactSubscriptionItemToggled(String subscriptionId, List<String> scopes, bool isSubscribed) {
+    ScopedSubscriptionListEditor editor = Airship.editContactSubscriptionLists();
+    if (isSubscribed) {
+      editor.subscribe(subscriptionId, scopes);
+      activeContactSubscriptions.putIfAbsent(subscriptionId, () => scopes);
+    } else {
+      editor.unsubscribe(subscriptionId, scopes);
+      activeContactSubscriptions.remove(subscriptionId);
+    }
+    editor.apply();
+    //setState(() {});
```

Comment:
should this be uncommented or removed?

Reply:
Yes 👍  it should be uncommented. I did some other improvements if you want to take a look